lib/horizon/payment_paths.ex

```diff
@@ -0,0 +1,205 @@
+defmodule Stellar.Horizon.PaymentPaths do
+  @moduledoc """
+  Exposes functions to interact with Paths in Horizon.
+
+  You can:
+  * Lists the paths a payment can take based on the amount of an asset you want the recipient to receive.
+  * Lists the paths a payment can take based on the amount of an asset you want to send.
+
+  Horizon API reference: https://developers.stellar.org/api/aggregations/paths/
+  """
+
+  alias Stellar.Horizon.{Error, Paths, Request}
+
+  @type args :: Keyword.t()
+  @type opt :: atom()
+  @type path :: String.t()
+  @type resource :: Paths.t()
+  @type response :: {:ok, resource()} | {:error, Error.t()}
+
+  @endpoint "paths"
+
+  @doc """
+    List Payment Paths
+
+  ## Parameters
+    * `source_account`: The Stellar address of the sender.
+    * `destination_asset_type`: The type for the destination asset.
+    * `destination_amount`: The amount of the destination asset that should be received.
+
+  ## Options
+    * `destination_account`: The Stellar address of the reciever.
+    * `destination_asset_issuer`: The Stellar address of the issuer of the destination asset. Required if the `destination_asset_type` is not native
+    * `destination_asset_code`: The code for the destination asset. Required if the `destination_asset_type` is not native.
+
+  ## Examples
+
+    iex> PaymentPaths.list_paths(source_account: "GBRSLTT74SKP62KJ7ENTMP5V4R7UGB6E5UQESNIIRWUNRCCUO4ZMFM4C",
+                                 destination_asset_type: "native",
+                                 destination_amount: 5
+                                )
+    {:ok, %Paths{records: [%Path{}, ...]}}
+
+    # list with `destination_account`
+    iex> PaymentPaths.list_paths(source_account: "GBRSLTT74SKP62KJ7ENTMP5V4R7UGB6E5UQESNIIRWUNRCCUO4ZMFM4C",
+                                 destination_asset_type: "native",
+                                 destination_amount: 5,
+                                 destination_account: "GBRSLTT74SKP62KJ7ENTMP5V4R7UGB6E5UQESNIIRWUNRCCUO4ZMFM4C"
+                                )
+    {:ok, %Paths{records: [%Path{}, ...]}}
+
+    # list with more options
+    iex> PaymentPaths.list_paths(source_account: "GBRSLTT74SKP62KJ7ENTMP5V4R7UGB6E5UQESNIIRWUNRCCUO4ZMFM4C",
+                                 destination_asset_type: "credit_alphanum4",
+                                 destination_amount: 5,
+                                 destination_account: "GBRSLTT74SKP62KJ7ENTMP5V4R7UGB6E5UQESNIIRWUNRCCUO4ZMFM4C",
+                                 destination_asset_code: "TEST",
+                                 destination_asset_issuer: "GA654JC6QLA3ZH4O5V7X5NPM7KEWHKRG5GJA4PETK4SOFBUJLCCN74KQ"
+                                )
+    {:ok, %Paths{records: [%Path{}, ...]}}
+  """
+
+  @spec list_paths(args :: args()) :: response()
+  def list_paths(args \\ []) do
+    :get
+    |> Request.new(@endpoint)
+    |> Request.add_query(args, extra_params: allowed_query_options(:opt1))
```

Comment:
@MrJuanGaviriaK it is not really clear what `opt1` means. Could you please use declarative names instead?

Reply:
I'm going to fix it, thanks!!!